src/pages/index.js

```diff
@@ -3,104 +3,248 @@ import FormValidator from "../components/FormValidator.js";
-import { config } from "../utils/Constants.js";
-
-console.log(initialCards);
+import {
+  profileEditButton,
+  avatarUpdateButton,
+  addCardButton,
+  profileTitleInput,
+  profileDescriptionInput,
+  avatarUrlInput,
+  profileEditForm,
+  addCardForm,
+  updateAvatarForm,
+  cardSelector,
+} from "../utils/constants.js";
+import { config } from "../utils/constants.js";
+import Api from "../components/Api.js";
 
 // Linked classes
+
 const popupWithEditProfileForm = new PopupWithForm(
   {
     popupSelector: "#profile-edit-modal",
   },
-  handleProfileEditSubmit,
+  handleProfileEditSubmit
+);
+
+const popupWithEditAvatarForm = new PopupWithForm(
+  {
+    popupSelector: "#update-avatar-modal",
+  },
+  handleAvatarUpdateSubmit
 );
 
 const popupWithAddCardForm = new PopupWithForm(
   {
     popupSelector: "#add-card-modal",
   },
-  handleAddCardFormSubmit,
+  handleAddCardFormSubmit
 );
 
-const popupWithImage = new PopupWithImage({popupSelector: "#preview-image-modal",});
+const popupWithImage = new PopupWithImage({
+  popupSelector: "#preview-image-modal",
+});
 
-const section = new Section(
+const popupWithDeleteButton = new PopupWithConfirm(
   {
-    items: initialCards,
-    renderer: renderCard,
+    popupSelector: "#delete-card-modal",
   },
-  ".cards__list"
+  openConfirmPopup
 );
 
 const userInfo = new UserInfo({
-  nameElement: ".profile__title",
-  jobElement: ".profile__description",
+  nameElement: ".profile__name",
+  jobElement: ".profile__about",
+  avatarElement: ".profile__avatar",
 });
 
 popupWithEditProfileForm.setEventListeners();
 popupWithAddCardForm.setEventListeners();
 popupWithImage.setEventListeners();
-
-
-
+popupWithDeleteButton.setEventListeners();
+popupWithEditAvatarForm.setEventListeners();
 
 /*Function*/
 
 function renderCard(item, method = "addItem") {
-    const cardElement = getCardElement(item);
-    section[method](cardElement);
+  const cardElement = getCardElement(item);
+  section[method](cardElement);
 }
 
 function getCardElement(cardData) {
-  const card = new Card(cardData, cardSelector, handleImageClick);
+  const card = new Card(
+    cardData,
+    cardSelector,
+    (cardInstance) => openConfirmPopup(cardInstance),
+    handleImageClick,
+    handleLikeButton
+  );
   return card.getView();
 }
 
 function handleImageClick(data) {
-    popupWithImage.open({ name: data.name, link: data.link });
+  console.log(data.name);
+  popupWithImage.open({ name: data.name, link: data.link });
+}
+
+function handleLikeButton(cardId, likeButton) {
+  if (!cardId) return;
+  const isLiked = likeButton.classList.contains("card__like-button_active");
+
+  api
+    .toggleLike(cardId, isLiked)
+    .then((updatedCard) => {
+      if (updatedCard && "isLiked" in updatedCard) {
+        likeButton.classList.toggle(
+          "card__like-button_active",
+          updatedCard.isLiked
```

Comment:
Now only the `Card` class knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle